src/main/webapp/js/calendar.js

```diff
@@ -13,22 +13,187 @@
+  }
+
+  /**
+   * Adds a course to the calendar given the day of the week and a time
+   * @param {Object} course The JSON Object for the course
+   * @param {string} startTime The string representation of a 12-hour clock
+   *     time eg. 8:30pm
+   * @param {string} endTime The string representation of a 12-hour clock
+   *     time eg. 8:30pm
+   * @param {enum_days} day The day that the course falls on
+   */
+  function addCourseToCalendar(course, startTime, endTime, day) {
+    const startDate = createDateFromTimeString(startTime, day);
+    const endDate = createDateFromTimeString(endTime, day);
+    createSchedule(course, startDate, endDate);
+  }
+
+  /**
+   * Creates a schedule on the calendar using the toast API
+   * @param {Object} course The JSON Object for the course
+   * @param {Date} startDate The Date object of the start time
+   * @param {Date} endDate The Date object of the end time
+   */
+  function createSchedule(course, startDate, endDate) {
+    // TODO(savsa): Have each schedule be a different color. The color is
+    // currently hard coded to be blue so all schedules look the same.
+    cal.createSchedules([{
+      id: id.toString(),
+      color: '#ffffff',
+      bgColor: '#00a9ff',
+      borderColor: '#00a9ff',
+      calendarId: '1',
+      title: course.name,
+      category: 'time',
+      location: course.building,
+      start: startDate,
+      end: endDate,
+      isReadOnly: true,
+    }]);
+    // Each schedule needs a unique ID. We'll start at 1 and go up from there.
+    id++;
+  }
+
+  /**
+   * Creates a schedule on the calendar using the toast API
+   * @param {enum_days} day The day that the course falls on
+   * @param {int} hour The army-time hour portion of the time/date we want
+   *     to create
+   * @param {hour} min The army-time minute portion of the time/date we want
+   *     to create
+   * @return {Date} The new Date object
+   */
+  function createDate(day, hour, min) {
+    const monday = new Date('January 3, 2000 00:00');
+    const rightDay = new Date(monday.getTime());
+    rightDay.setDate(monday.getDate() + day);
+    return addTimeToDate(rightDay, hour, min);
+  }
+
+  /**
+   * Adds time to a Date object
+   * @param {Date} day The Date object
+   * @param {int} hour The army-time amount of hours we want to add
+   * @param {hour} min The army-time amount of minutes we want to add
+   * @return {Date} The new Date object
+   */
+  function addTimeToDate(date, hours, mins) {
+    const newDate = new Date(date.getTime());
+    newDate.setHours(date.getHours() + hours);
+    newDate.setMinutes(date.getMinutes() + mins);
+    return newDate;
+  }
+
+  /**
+   * Creates a Date object from the string representation of a time
+   * @param {string} time The string representation of a 12-hour clock time eg.
+   *     8:30pm
+   * @param {enum_days} day The day that the course falls on
+   * @return {Date} The new Date object
+   */
+  function createDateFromTimeString(time, day) {
```

Comment:
consider using moment js to parse, instead of writing your own logic: https://stackoverflow.com/a/34540549